app/src/androidTest/java/net/osmtracker/layouts/RepositorySettingsDialogTest.java

```diff
@@ -0,0 +1,112 @@
+package net.osmtracker.layouts;
+
+import android.support.test.espresso.ViewAssertion;
+import android.support.test.rule.ActivityTestRule;
+import android.view.View;
+
+import net.osmtracker.OSMTracker;
+import net.osmtracker.R;
+import net.osmtracker.activity.AvailableLayouts;
+import net.osmtracker.util.TestUtils;
+
+import org.hamcrest.Matcher;
+import org.junit.Rule;
+import org.junit.Test;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.clearText;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.assertion.ViewAssertions.doesNotExist;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isChecked;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.isEnabled;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static net.osmtracker.util.TestUtils.checkToastIsShownWith;
+import static net.osmtracker.util.TestUtils.getStringResource;
+import static org.hamcrest.core.IsNot.not;
+
+public class RepositorySettingsDialogTest {
+
+    @Rule
+    public ActivityTestRule<AvailableLayouts> mRule = new ActivityTestRule<>(AvailableLayouts.class);
+
+
+    @Test
+    public void testToggleBehaviour(){
+        onView(withId(R.id.github_config)).perform(click());
+
+        onView(withId(R.id.default_server)).perform(click());
+        checkStateAfterToggle(R.id.default_server, R.id.custom_server);
+        checkTextFieldsState(not(isEnabled()));
+        checkTextFieldsDefaultValues();
+
+        onView(withId(R.id.custom_server)).perform(click());
+        checkStateAfterToggle(R.id.custom_server, R.id.default_server);
+        checkTextFieldsState(isEnabled());
+    }
+
+    @Test
+    public void testRepositoryValidation(){
```

Comment:
Isn't better to use the constant values for github username, repository name and branch name, declared in OSMTracker.Preferences. (Except in line 53 where the "NONE" in requiered).